pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,434 @@
+)
+
+const (
+	controllerName = "AntreaAgentPodController"
+	minRetryDelay  = 2 * time.Second
+	maxRetryDelay  = 120 * time.Second
+	numWorkers     = 4
+)
+
+const (
+	networkAttachDefAnnotationKey = "k8s.v1.cni.cncf.io/networks"
+	cniPath                       = "/opt/cni/bin/"
+	sriovInterfaceType            = "sriov"
+	defaultSecondaryInterfaceName = "eth1"
+	start_iface_index             = 1
+	end_iface_index               = 10
+	max_rand_index                = 25
+)
+
+// Set resyncPeriod to 0 to disable resyncing.
+const resyncPeriod = 0 * time.Minute
+
+type PodController struct {
+	kubeClient         clientset.Interface
+	netAttachDefClient netdefclient.K8sCniCncfIoV1Interface
+	queue              workqueue.RateLimitingInterface
+	podInformer        cache.SharedIndexInformer
+	podLister          corelisters.PodLister
+	nodeName           string
+	podCache           cnipodcache.CNIPodInfoStore
+	cniServer          *cniserver.CNIServer
+}
+
+func NewPodController(kubeClient clientset.Interface,
+	netAttachDefClient netdefclient.K8sCniCncfIoV1Interface,
+	podInformer cache.SharedIndexInformer,
+	resyncPeriod time.Duration,
+	nodeName string,
+	podCache cnipodcache.CNIPodInfoStore,
+	cniServer *cniserver.CNIServer) *PodController {
+
+	pc := PodController{
+		kubeClient:         kubeClient,
+		netAttachDefClient: netAttachDefClient,
+		queue:              workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "podcontroller"),
+		podInformer:        podInformer,
+		podLister:          corelisters.NewPodLister(podInformer.GetIndexer()),
+		nodeName:           nodeName,
+		podCache:           podCache,
+		cniServer:          cniServer,
+	}
+
+	podInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    pc.enqueuePod,
+			DeleteFunc: pc.enqueuePod,
+			UpdateFunc: func(old, cur interface{}) { pc.enqueuePod(cur) },
+		},
+		resyncPeriod,
+	)
+	return &pc
+}
+
+func podKeyGet(pod *corev1.Pod) string {
+	return pod.Namespace + "/" + pod.Name
+}
+
+func generatePodSecondaryIfaceName(podCNIInfo *cnipodcache.CNIConfigInfo) string {
+	// Assign default interface name, if podCNIInfo.NetworkConfig is empty.
+	if count := len(podCNIInfo.NetworkConfig); count == 0 {
+		return defaultSecondaryInterfaceName
+	} else {
+		// Generate new interface name (eth1,eth2..eth10) and return to caller.
+		for ifaceIndex := start_iface_index; ifaceIndex < end_iface_index; ifaceIndex++ {
+			ifName := fmt.Sprintf("%s%d", "eth", ifaceIndex)
+			_, exist := podCNIInfo.NetworkConfig[ifName]
+			if !exist {
+				return ifName
+			}
+		}
+	}
+	// Generates random interface name and return. Above execution will try to ensure allocating interface names in order.
+	// If none available between eth1 to eth10 (already used per Pod), generate random name with the integer range.
+	return string("eth") + strconv.Itoa(rand.IntnRange(end_iface_index, max_rand_index))
```

Comment:
Maybe this was clarified before, but I am confused as to why there is this random fallback. Why not use a much larger value for `end_iface_index`?

Reply:
missed to reply on this comment earlier. We support 10 interfaces (eth1 to eth10), if none of them are available we allow to generate name b/w eth11 to eth25 (to get the interface names generated in a range and to avoid random names). Ideally this should never be hitting in any of our known uses cases so far.